### PR TITLE
docs: improve README file

### PR DESCRIPTION
**Pull request by peter-rr**
Adding a complete version of the README file, so we can start with the donation process of this repo.

**Related issue(s)**
Relates to #16

### CODE REVIEW

**Comment by smoya** [comment]
We require all PRs to follow [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/). 
More details 👇🏼
```
 The subject "Improve README file" found in the pull request title "docs: Improve README file" should start with a lowercase character.

```

**Comment by peter-rr** [comment]
@smoya Feel free to add any kind of suggestion and let me know if anything missing on the document :)

File: README.md [hunk 1]
```diff
@@ -1,3 +1,41 @@
...
+
+It aids in comprehending how AsyncAPI tools are used and adopted, facilitating ongoing improvements to our specifications and tools. Currently is being used by @asyncapi/cli as a POC and next step will be making use of it on @asyncapi/studio as well.
+
 Requires NodeJS v18 or greater.
+
+## Metrics collected
+We are collecting the following metrics:
+
+- `asyncapi_adoption.action.invoked`:
+With this metric we are tracking the command executed on the CLI as soon as the command is invoked, so it hasn't been completely executed and not finished yet. We just want to know which commands are used, regardless they have failed or succeeded.
```
Comment (smoya) [inline README.md]:
```suggestion
With this metric we are tracking any action started (but not finished) in a tool. For example, a command that got executed on the [CLI](https://github.com/asyncapi/cli/) but didn't finish yet. We just want to know which commands are used, regardless they have failed or succeeded.
```

File: README.md [hunk 2]
```diff
@@ -1,3 +1,41 @@
...
+
+- `asyncapi_adoption.action.finished`:
+This metric tracks the command executed once it has already finished, carrying the result of the execution and some metadata based on the AsyncAPI document in place.
```
Comment (smoya) [inline README.md]:
```suggestion
With this metric we are trcking the action executed once it has already finished, carrying the result of the execution and some metadata. 
```

File: README.md [hunk 3]
```diff
@@ -1,3 +1,41 @@
...
+
+## Examples
+Example for `validate` command after being invoked:
```
Comment (smoya) [inline README.md]:
```suggestion
Example for[CLI](https://github.com/asyncapi/cli/) `validate`command after being invoked:
```

File: README.md [hunk 4]
```diff
@@ -1,3 +1,41 @@
...
+  asyncapi_messages: 3,
+  asyncapi_operations_send: 3,
+  asyncapi_operations_receive: 1,
+  asyncapi_schemas: 52,
+  action: 'validate'
+  }       1
+```
+
+## Data storage
+We are making use of [New Relic API](https://docs.newrelic.com/docs/apis/intro-apis/introduction-new-relic-apis/#rest-api) to send the metrics collected to the _New Relic_ servers, where they are stored, and finally visualized on our own dashboards at the AsyncAPI website.
```
Comment (smoya) [inline README.md]:
```suggestion
We are making use of [New Relic API](https://docs.newrelic.com/docs/apis/intro-apis/introduction-new-relic-apis/#rest-api) to send the metrics collected to the _New Relic_ servers, where they are stored, and finally visualized on our own dashboards at the AsyncAPI website (WIP).
```

Comment (smoya) [reply README.md]: Worth to add a section about data privacy, mentioning all data is anonymized and sensitive data will never be shared with us.

**Comment by smoya** [comment]
Missing the following sections:

- Installation
- Usage (with examples)
- Contributing

File: README.md (hunk 4 above)
Comment (peter-rr) [reply README.md]: Agree! Adding that section to the document 👍

File: README.md [hunk 5]
```diff
@@ -1,3 +1,52 @@
...
+We are making use of [New Relic API](https://docs.newrelic.com/docs/apis/intro-apis/introduction-new-relic-apis/#rest-api) to send the metrics collected to the _New Relic_ servers, where they are stored, and finally visualized on our own dashboards at the AsyncAPI website (WIP).
+
+## Data privacy
+All the data is anonymized, ensuring no sensitive data reaches our servers.
+
+Even though metrics collection is enabled by default, you can always disable tracking if you wish. To disable tracking, please run the following command:
+`asyncapi config analytics --disable`
+
+Once disabled, if you want to enable tracking back again then run:
+`asyncapi config analytics --enable`
```
Comment (peter-rr) [inline README.md]: @smoya Not sure if all this info would be necessary since it's just related to CLI use case 🤔

Comment (smoya) [reply README.md]: I don't thing it should be here. I would say we should add a statement mentioning all tools should enable a mechanism for disabling metrics.

Comment (peter-rr) [reply README.md:61]: Fair enough 👍 Gonna change it then.

**Comment by peter-rr** [comment]
> Missing the following sections:
> 
> * Installation
> * Usage (with examples)
> * Contributing

Missing sections already added. Ready for review 🚀

**Pull request merged**